docs/plugins/reindex.asciidoc

```diff
@@ -0,0 +1,608 @@
+[[plugins-reindex]]
+=== Reindex Plugin
+
+The reindex plugin adds two APIs:
+
+* `_update_by_query` updates all documents matching a query in place.
+* `_reindex` copies documents from one index to another.
+
+These APIs are brothers so they live in the same plugin. Both use
+{ref}/search-request-scroll.html[Scroll] and {ref}/docs-bulk.html[Bulk] APIs
+to send an index request per document though they may not work that way in the
+future. Regardless of how they work internally they will always work **as**
+**though** they send an index request per document that matches a scroll.
+
+[float]
+==== Installation
+
+This plugin can be installed using the plugin manager:
+
+[source,sh]
+----------------------------------------------------------------
+sudo bin/plugin install reindex
+----------------------------------------------------------------
+
+The plugin must be installed on every node in the cluster, and each node must
+be restarted after installation.
+
+[float]
+==== Removal
+
+The plugin can be removed with the following command:
+
+[source,sh]
+----------------------------------------------------------------
+sudo bin/plugin remove reindex
+----------------------------------------------------------------
+
+The node must be stopped before removing the plugin.
+
+[[update-by-query-usage]]
+==== Using `_update_by_query`
+
+The simplest usage of `_update_by_query` just `touch`es every document in an
```

Comment:
This command goes a bit further than just "touching" a file (updating access time).
Here, "touching" can make changes to the indexed fields if mapping has been extended since (example you give below). As explanation, I could imagine something along the lines of "reindexing as-is all documents available at the time the command was started". But then, what about conflicts / versioning? "just touches" is a huge understatement of what's going on. My point here was just that I would prefer to have more details / motivation on what this command tries to achieve. I know that you care more about explaining the syntax in this paragraph. I just wonder if that specific command has any practical use. If so, we could provide a motivation for it (splitting hairs, I know ;-)
.